gedi/process_l2a.py

```diff
@@ -227,57 +273,155 @@ def df_to_geojson(df, outfile):
     GeoDF = GeoDF.drop(columns=["lat_lowestmode", "lon_lowestmode"])
     GeoDF.to_file(outfile, driver="GeoJSON")
 
+def download_url(url : str, dir : str, chunk_size : int = 128) -> None:
+    """
+    Download the zip files generated by NASA EarthData Search.
+    Parameters
+    ----------
+    url : str
+        URL used to download data zip package
+    dir : str
+        Directory used to store the zip file and unzipped contents
+    chunk_size : int
+        Determines the size of each chunk written to file during data streaming
+    Side Effects
+    -------
+    Downloads and unpacks the zip file from the provide url
+    Returns
+    -------
+    None
+    """
+    filepath = os.path.join(dir, "granuleData.zip")
+    r = requests.get(url, stream = True)
```

Comment:
I think the way this does the checking and prints a descriptive error to the user is useful, especially since we want this function to return a bool, rather than throw the exception outright. 

But, for future reference, a helpful tip that someone taught me: in most cases, using `raise_for_status` (i.e. `r.raise_for_status()` in this case) does the sensible thing you'd want when checking a response. It raises an error if the request returns an error or times out, and it returns none otherwise. Again, I think that _is not_ actually what we want here, but it's worth having that in your toolbox for the times where it is what you want to do.